protostar/starknet_gateway/multicall/multicall_structs.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+from typing import Optional, Union
+
+from protostar.starknet import Address, Selector
+
+DeployCallName = str
+
+MulticallInputCalldata = list[Union[int, DeployCallName]]
+
+
+@dataclass(frozen=True)
+class CallBase:
+    calldata: MulticallInputCalldata
+
+
+@dataclass(frozen=True)
+class InvokeCall(CallBase):
+    address: Union[DeployCallName, Address]
+    selector: Selector
+
+
+@dataclass(frozen=True)
+class DeployCall(CallBase):
+    class_hash: int
+    name: Optional[DeployCallName] = None
+
+
+@dataclass
+class ResolvedCall:
+    address: Address
+    selector: Selector
+    calldata: list[int]
```

Comment:
Isn't the structure basically same as the `Call` from starknet.py?

Reply:
It's similar. This data structure uses value objects. Use case layer shouldn't be polluted by external library.